bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/ReadRowsRequestManager.java

```diff
@@ -67,77 +47,18 @@
    * <p>
    * Constructor for ResumingStreamingResultScanner.
    * </p>
-   * @param retryOptions a {@link RetryOptions} object.
    * @param originalRequest a {@link ReadRowsRequest} object.
-   * @param rpcMetrics a {@link com.google.cloud.bigtable.grpc.async.BigtableAsyncRpc.RpcMetrics} object to keep track of retries and
-   *          failures.
-   * @param logger a {@link Logger} to log info messages about the state of retries.
    */
-  public ReadRowsRequestRetryHandler(RetryOptions retryOptions, ReadRowsRequest originalRequest,
-      RpcMetrics rpcMetrics, Logger logger) {
+  public ReadRowsRequestManager(ReadRowsRequest originalRequest) {
     this.originalRequest = originalRequest;
-    this.retryOptions = retryOptions;
-    this.rpcMetrics = rpcMetrics;
-    this.logger = logger;
-  }
-
-  public void update(FlatRow result) {
-    updateLastFoundKey(result.getRowKey());
-    rowCount++;
-    // We've had at least one successful RPC, reset the backoff and retry counter
-    currentErrorBackoff = null;
-    timeoutRetryCount = null;
   }
 
-  void updateLastFoundKey(ByteString key) {
+  public void updateLastFoundKey(ByteString key) {
     this.lastFoundKey = key;
+    rowCount++;
```

Comment:
Would it be make sense to add a sanity check that the key is >= than the previous lastFoundKey?